server/routes/api/v1/allergies/index.js

```diff
@@ -0,0 +1,41 @@
+import { Role } from '../../../../models/user.js';
+
+export default async function (fastify) {
+  fastify.get(
+    '',
+    {
+      schema: {
+        querystring: {
+          allergy: { type: 'string' },
+        },
+      },
+      onRequest: fastify.requireUser([Role.ADMIN, Role.STAFF, Role.VOLUNTEER]),
+    },
+    async (request, reply) => {
+      const allergy = request.query.allergy.trim();
+
+      // prevent empty string from returning all allergies
+      if (!allergy.length) {
+        return reply.send({ message: 'No query provided' });
+      }
+
+      const NO_ALLERGIES = ['none', 'no allergies', 'no known allergies', 'n/a'];
+
+      if (NO_ALLERGIES.includes(allergy.toLowerCase())) {
+        return reply.send({ message: 'No known allergies' });
+      }
+
+      const results = await fastify.prisma.allergy.findMany({
+        orderBy: [{ name: 'asc' }],
+        where: { name: { contains: allergy, mode: 'insensitive' } },
+        select: { name: true },
+      });
+
+      if (!results.length) {
+        return reply.notFound();
+      }
```

Comment:
I wasn't sure about how to handle a situation where an unknown but valid allergy/condition/medication is provided, so for now I used a `NOT FOUND` error for this iteration.